pkg/cmd/pipelinerun/log_reader.go

```diff
@@ -121,11 +129,35 @@ func (lr *LogReader) readAvailableLogs(pr *v1alpha1.PipelineRun) (<-chan Log, <-
 
 			pipeLogs(logC, errC, tlr)
 		}
+		if pr.Status.Conditions[0].Status == corev1.ConditionFalse {
+			errC <- fmt.Errorf(pr.Status.Conditions[0].Message)
+		}
 	}()
 
 	return logC, errC, nil
 }
 
+func (lr *LogReader) waitUntilAvailable(timeout int, limit int) {
+	if timeout > 0 {
+		tkn := lr.Clients.Tekton
+		run, err := tkn.TektonV1alpha1().PipelineRuns(lr.Ns).Get(lr.Run, metav1.GetOptions{})
+		if err != nil {
+			fmt.Println(err.Error())
+		}
+		for _, tr := range run.Status.TaskRuns {
+			if tr.Status.Steps != nil && tr.Status.Steps[0].ContainerState.Waiting != nil {
+				if timeout == limit {
+					fmt.Println("Pipeline still running ...")
+				}
+				time.Sleep(1 * time.Second)
```

Comment:
Instead of using `time.Sleep`, we could use the `time` `Ticker` construction and `time.Duration` for the timeout :angel:

Reply:
in new code, we are using Watch api.